docs/index.md

```diff
@@ -1313,25 +1318,160 @@ Statement 3 looks suspiciously like axiom `ax-2`, which states:
+but for now we'll need to work around this.
+
+With a little extra work we can give metamath-lamp the information it needs.
+Let's assume that we know we want to use `ax-2` to prove this.
 
-We can help metamath-lamp along, however. We just need to replace the
+One approach would be replace the
 work variable "&amp;W1" with the expression required by `ax-2`, namely,
 `( ph -> ( ps -> ch ) )`.
+Here's how you could do that, but note that we're going to cancel
+instead of completing this step:
 
 > Select the icon "A with arrow" (that is, "replace").
 > In "Replace what" use the value &amp;W1 and in
 > "replace with" use the value `( ph -> ( ps -> ch ) )` and then press Return.
 > Click "Find Substitution" button; metamath-lamp will
 > show one possible substitution.
-> Click the "Apply" button to apply the substitution to all the statements.
-> Then press unify.
+> You *could* click on the "Apply button and then unify, but don't;
+> select Cancel instead.
+
+That would have proven step 3 using ax-2.
+However, if the expressions were more complex it might be hard to
+make sure we were connecting them the right way.
+
+<!-- The following is based on
+https://drive.google.com/file/d/1KIr0eOEmH4VoIHOHFhqXwBn08h-xGicV/view?usp=sharing -->
+
+A more general approach would be to add the step we want to use, and then
+perform substitutions until we can unify them.
+This is a better approach for more complicated situations, so let's see it.
+First, let's bring in a step that uses the assertion we wish to use
+(in this case `ax-2`):
+
+> Select the magnifying glass ("Search"), enter the label `ax-2` and
+> press "Search". Select `ax-2` and press "Choose Selected".
+
+We have added a new step 4, which uses `ax-2` but with work variables:
+
+~~~~
+|- ( ( &W3 -> ( &W4 -> &W2 ) ) -> ( ( &W3 -> &W4 ) -> ( &W3 -> &W2 ) ) )
+~~~~
+
+We are now going to modify steps 3 and 4 until metamath-lamp can
+unify them.
+The key feature we're going to use is that you can use the
+statement fragment selectors to *simultaneously* select two fragments,
+which may include *multiple* work variables, and
+then use "replace". Replace will use the two selected fragments, making
+this process *much* easier.
+
+> Use Alt+click to select, in step *4*, the last `->`.
+
+The statement fragment selector dialogue has appeared under step 4 and
+we now have this fragment selected:
+
+~~~~
+( ( &W3 -> &W4 ) -> ( &W3 -> &W2 ) )
+~~~~
+
+You could use the dialogue to
+increase or decrease the size of the fragment, but we don't need to.
+Now let's select the equivalent fragment in statement 3:
+
+> Use Alt+click to select, in step *3*, the third `->`.
+
+Another fragment selector dialogue has appeared under step 3 and
+it has this fragment selected:
+
+~~~~
+( ( ph -> ps ) -> ( ph -> ch ) )
+~~~~
+
+If you didn't pci the right fragment, use alt+click again to get the
+correct selection.
+
+**Note**: you *can* select two fragments at the same time.
```

Comment:
`**Note**: you *can* select two fragments at the same time.`

mm-lamp also supports selecting only one fragment. In that case only one text field in the "Replace" dialog will be filled in. Also, instead of fragment selection, you may use statement checkboxes with the same effect.

I am not sure where is the best place to mention this. So I just put a comment here.

Reply:
I'll mention it here, and expland on it in the reference section.

There's an oddity: if I understand things correctly, for fragments, the "first one listed" is first, but for statements, "the first one selected" is first. The rule should be the same (I prefer "first one selected). I'll make an issue noting this.

Reply:
ok, I will fix it.